src/engraving/dom/page.cpp

```diff
@@ -131,9 +140,42 @@ Text* Page::layoutHeaderFooter(int area, const String& ss) const
     case 5: align = { AlignH::RIGHT, AlignV::BOTTOM };
         break;
     }
+
+    text->mutldata()->blocks = { tb };
+
+    //! NOTE: Keep in sync with replaceTextMacros
+    std::wregex copyrightSearch(LR"(\$[cC])");
+    std::wregex pageNumberSearch(LR"(\$[pPnN])");
+
+    bool containsCopyright = ss.contains(copyrightSearch);
+    bool containsPageNumber = ss.contains(pageNumberSearch);
+
+    // TextBase properties (slight hack) - we'll use copyright/page number styling if the string contains
+    //copyright/page number macros (hack because any non-copyright text will also adopt these style values)
     text->setAlign(align);
-    text->setXmlText(s);
+    text->setTextLineSpacing(style().styleD(HEADERFOOTER_SID_FOR_SUFFIX(LineSpacing)));
+    text->setColor(style().styleV(HEADERFOOTER_SID_FOR_SUFFIX(Color)).value<Color>());
+
+    const bool spatiumDependent = style().styleB(HEADERFOOTER_SID_FOR_SUFFIX(FontSpatiumDependent));
+    const double factor = spatiumDependent ? SPATIUM20 : DPMM;
+    const PointF p = style().styleV(HEADERFOOTER_SID_FOR_SUFFIX(Offset)).value<PointF>();
+    const PointF pointScaled = { p.x() * factor, p.y() * factor };
+    text->setOffset(pointScaled);
+
+    // Frame properties...
+    text->setFrameType(style().styleV(HEADERFOOTER_SID_FOR_SUFFIX(FrameType)).value<FrameType>());
+    text->setFrameColor(style().styleV(HEADERFOOTER_SID_FOR_SUFFIX(FrameFgColor)).value<Color>());
+    text->setBgColor(style().styleV(HEADERFOOTER_SID_FOR_SUFFIX(FrameBgColor)).value<Color>());
+    text->setFrameWidth(style().styleV(HEADERFOOTER_SID_FOR_SUFFIX(FrameWidth)).value<Spatium>());
+    text->setPaddingWidth(style().styleV(HEADERFOOTER_SID_FOR_SUFFIX(FramePadding)).value<Spatium>());
+    text->setFrameRound(style().styleI(HEADERFOOTER_SID_FOR_SUFFIX(FrameRound)));
```

Comment:
Instead of all this, it might be possible to call `text->initTextStyleType(…)` with TextStyleType::COPYRIGHT / PAGE_NUMBER / HEADER / FOOTER conditionally